Sources/Packages.php

```diff
@@ -1356,6 +1356,10 @@ function PackageBrowse()
 	$context['page_title'] .= ' - ' . $txt['browse_packages'];
 
 	$context['forum_version'] = $forum_version;
+	$context['available_modification'] = array();
+	$context['available_avatar'] = array();
+	$context['available_language'] = array();
+	$context['available_unknown'] = array();
```

Comment:
After resolving conflicts, I suggest removing these four lines and instead using a foreach to loop through `$context['modification_types']` to set these variables. If you put that loop after the hook, then we'll be able to handle any types added by the hook, too.

Reply:
Eh,  like this?

```	call_integration_hook('integrate_modification_types');

	require_once($sourcedir . '/Subs-List.php');
	foreach ($context['modification_types'] as $type)
	{
		$context['available_' . $type] = array();

		// Use the standard templates for showing this.
		$listOptions = array(
```

Reply:
Yes.